deployment/promotion/promote.py

```diff
@@ -75,33 +79,49 @@ def auth_context(host, email, devkey, workspace):
 def post(auth, path, body):
     auth['headers']['Content-Type'] = 'application/json'
     body['workspaceName'] = auth['workspace']
-    res = requests.post("https://{}{}".format(auth["host"], path), headers=auth['headers'], json=body)
-    if res.ok:
-        return res.json()
-    raise Exception(res.text)
+    try:
+        res = requests.post("https://{}{}".format(auth["host"], path), headers=auth['headers'], json=body)
+        res.raise_for_status()
+    except requests.exceptions.RequestException as e:
+        if len(e.response.text) > 0:
```

Comment:
fyi! It'd be more idiomatic in Python to write

```suggestion
        if e.response.text:
```

but that's understandably confusing, so it's alright!